src/engine/Engine.cpp

```diff
@@ -1995,11 +1993,11 @@ void Engine::updateScores()
     }
 }
 
-PiecewiseLinearConstraint *Engine::pickSplitPLConstraint()
+PiecewiseLinearConstraint *Engine::pickSplitPLConstraint( DivideStrategy strategy )
 {
     _candidatePlConstraints.clear();
     ENGINE_LOG( Stringf( "Picking a split PLConstraint..." ).ascii() );
-    updateScores();
+    updateScores( strategy );
     ENGINE_LOG( Stringf( "Done updating scores..." ).ascii() );
     if ( _candidatePlConstraints.empty() )
     {
```

Comment:
There's something here I'm not following. When the SMT core decides it's time to split, it calls Engine::pickSPlitPLConstraints. The engine then computes scores for the various constraints, according to the current strategy. However, where is the step that picks the constraint with the **highest** score? It seems we're just picking the first constraint in _candidatePlConstraints?

Reply:
Since _candidatePLConstraints is a Set which uses std::set as a container, which is sorted according to its specification. On the other hand, the comparator of PiecewiseLinearConstraint is overloaded with comparing the score.